tests/e2e_tests/test_submit.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+from playwright.sync_api import Page, expect
+
+from tests.e2e_tests.helpers import (
+    create_user_with_roles,
+    generate_email_address,
+    has_sent_la_confirmation_email,
+    login_via_magic_link,
+)
+from tests.e2e_tests.pages.submit import SubmitReportPage
+
+pytestmark = pytest.mark.e2e
+
+
+def test_submit_report(domains, authenticator_fund_config, page: Page):
+    email_address = generate_email_address(
+        test_name="test_submit_report",
+        email_domain="communities.gov.uk",
+    )
+
+    create_user_with_roles(
+        email_address,
+        ["PF_MONITORING_RETURN_SUBMITTER", "TF_MONITORING_RETURN_SUBMITTER"],
+    )
+
+    login_via_magic_link(
+        page,
+        email_address=email_address,
+        domains=domains,
+        fund_config=authenticator_fund_config,
+    )
+
+    request_data_page = SubmitReportPage(page, domain=domains.submit)
+    request_data_page.navigate()
+
+    expect(page.get_by_role("heading", name="Submit monitoring and evaluation data dashboard")).to_be_visible()
+
+    request_data_page.select_fund("Pathfinders")
```

Comment:
I think the approach with the `SubmitReportPage` class is that it represents a specific page on Submit - rather than all of the pages. So when you click the Pathfinders link, that takes you to a new URL, which is a new page, so we should probably return another (different) Page instance from here and then keep using that.

See also https://playwright.dev/python/docs/pom

So maybe `select_fund` should return a `SubmitUploadPage` instance, which has the `submit_report` method, which returns a `SubmitUploadConfirmationPage` instance (or maybe a `SubmitUploadErrorsPage` instance if the upload has errors).

Reply:
I made so far part of this change. 

We have a separate class per page. I even created classes for the magic link journey. 

Classes are not yet returning a different instance when we redirect. I am still playing with the idea.. I will push an update soon.

Reply:
I might have overcooked it.

Added initial error upload journey and general error upload journey. It's ready for review.